src/plugins/redirectYouTube.desktop/index.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Vencord, a Discord client mod
+ * Copyright (c) 2024 Vendicated and contributors
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+import { definePluginSettings } from "@api/Settings";
+import { Devs } from "@utils/constants";
+import definePlugin, { OptionType } from "@utils/types";
+
+export default definePlugin({
+    name: "RedirectYouTube",
+    description: "Replace all YouTube embeds with an embed from an third-party YouTube frontend, such as Invidious or Piped.",
+    authors: [Devs.katlyn],
+    settings: definePluginSettings({
+        instances: {
+            type: OptionType.STRING,
+            description: "A comma-separated list of alternate YouTube frontends (including protocol such as https). If multiple frontends are specified, one will be chosen at random.",
+            restartNeeded: true,
+            default: "https://invidious.fdn.fr/, https://vid.puffyan.us/, https://invidious.nerdvpn.de/, https://invidious.projectsegfau.lt/, https://invidious.flokinet.to/"
+        }
+    }),
+    patches: [{
+        find: "case\"YouTube\":",
+        replacement: {
+            match: /(let{src:\i,autoMute:\i.{10,200}src:)(\i)/,
```

Comment:
if you don't actually need to match it, just use a lookbehind
```suggestion
            match: /(?<=let{src:\i,autoMute:\i.{10,200}src:)(\i)/,
```